fastlane_core/spec/helper_spec.rb

```diff
@@ -58,6 +58,30 @@
       end
     end
 
+    describe "#fs_is_insensitive?" do
+      it "detects fs sensitivity" do
+        expect(File).to receive(:exist?).and_return(true)
+        expect(FastlaneCore::Helper.fs_is_insensitive?).to be true
+        expect(File).to receive(:exist?).and_return(false)
+        expect(FastlaneCore::Helper.fs_is_insensitive?).to be false
+      end
+    end
+
+    describe "#compare_paths?" do
+      it "Compare Path on sensitive system" do
```

Comment:
from a rspec style perspective, you could use `context` to avoid having to `expect` multiple times and make the intent clearer.

```
      context 'when the file system is insentive' do
        before(:each) do
          allow(FastlaneCore::Helper).to receive(:fs_is_insensitive?) { true }
        end
        it 'ignores case while comparing' do
             expect(FastlaneCore::Helper.compare_paths("FASTLANE", "fastlane")).to be true
        end
        it 'matches equal strings' do
             expect(FastlaneCore::Helper.compare_paths("fastlane", "fastlane")).to be true
        end
```

Not a big deal here, but might be useful in other "contextes" :)